app/src/test/kotlin/compiler/e2e/CorrectProgramsE2eTest.kt

```diff
@@ -314,4 +314,18 @@ class CorrectProgramsE2eTest {
             """
         )
     }
+
+    @Test
+    fun `test looping over arrays`() {
```

Comment:
Also at least one integration test (in the `tests/` folder) would be useful.

Reply:
You mean in `AstFactoryTest` for example?

Reply:
A test to check if a correct AST is produced (in `AstFactoryTest`) would also be welcome, but what I mean is a simple integration test which checks that the loop actually works when running the program (in the `tests/` folder in the project root).

Reply:
Will add tomorrow, after I decode how to do it.